at org.postgresql.jdbc2.AbstractJdbc2Connection.checkClosed(AbstractJdbc2Connection.java:714)
at org.postgresql.jdbc3.AbstractJdbc3Connection.createStatement(AbstractJdbc3Connection.java:230)
at org.postgresql.jdbc2.AbstractJdbc2Connection.createStatement(AbstractJdbc2Connection.java:191)
at com.mchange.v2.c3p0.impl.NewProxyConnection.createStatement(NewProxyConnection.java:96)
at com.adventnet.nms.db.util.DBXmlUtility.execute(DBXmlUtility.java:283)
at com.adventnet.nms.db.util.DBXmlUtility.getTableName(DBXmlUtility.java:3275)
---
com.adventnet.nms.store.NmsStorageException: This connection has been closed.
        at com.adventnet.nms.db.util.DBXmlUtility.execute(DBXmlUtility.java:299)
        at com.adventnet.nms.db.util.DBXmlUtility.modifyNode(DBXmlUtility.java:1390)
        at com.adventnet.nms.mapdb.MapHandler.handleTreeUpdate(MapHandler.java:416)
        at com.adventnet.nms.mapdb.MapHandler.updateMap(MapHandler.java:390)
---
java.lang.NullPointerException
        at com.adventnet.nms.db.util.DBXmlUtility.updatePropTable(DBXmlUtility.java:3200)
        at com.adventnet.nms.db.util.DBXmlUtility.modifyOneNode(DBXmlUtility.java:1467)
        at com.adventnet.nms.db.util.DBXmlUtility.modifyNode(DBXmlUtility.java:1424)
        at com.adventnet.nms.mapdb.MapHandler.handleTreeUpdate(MapHandler.java:416)
        at com.adventnet.nms.mapdb.MapHandler.updateMap(MapHandler.java:390)
---
at com.adventnet.nms.db.util.DBConnectionUtil.refreshConnection(DBConnectionUtil.java:106)
at com.adventnet.nms.db.util.DBXmlUtility.repairConnection(DBXmlUtility.java:3717)
at com.adventnet.nms.db.util.DBXmlUtility.executeUpdate(DBXmlUtility.java:247)
at com.adventnet.nms.db.util.DBXmlUtility.updatePropTable(DBXmlUtility.java:3228)
at com.adventnet.nms.db.util.DBXmlUtility.modifyOneNode(DBXmlUtility.java:1467)
at com.adventnet.nms.db.util.DBXmlUtility.modifyNode(DBXmlUtility.java:1424)
at com.adventnet.nms.mapdb.MapHandler.handleTreeUpdate(MapHandler.java:416)
---
com.adventnet.nms.store.NmsStorageException: This connection has been closed.
        at com.adventnet.nms.db.util.DBXmlUtility.execute(DBXmlUtility.java:299)
        at com.adventnet.nms.db.util.DBXmlUtility.isNodeExist(DBXmlUtility.java:2678)
---
com.adventnet.nms.store.NmsStorageException: This connection has been closed.
        at com.adventnet.nms.db.util.DBXmlUtility.execute(DBXmlUtility.java:299)
        at com.adventnet.nms.db.util.DBXmlUtility.modifyNode(DBXmlUtility.java:1390)
---
[20 May 2016 05:16:38:701] SYS_ERR: [WARNING] [c3p0] Another error has occurred [ org.postgresql.util.PSQLException: This connection has been closed. ] which will not be reported to listeners!org.postgresql.util.PSQLException: This connection has been closed.
        at org.postgresql.jdbc2.AbstractJdbc2Connection.checkClosed(AbstractJdbc2Connection.java:714)
        at org.postgresql.jdbc3.AbstractJdbc3Connection.createStatement(AbstractJdbc3Connection.java:230)
        at org.postgresql.jdbc2.AbstractJdbc2Connection.createStatement(AbstractJdbc2Connection.java:191)
        at com.mchange.v2.c3p0.impl.NewProxyConnection.createStatement(NewProxyConnection.java:96)
        at com.adventnet.nms.db.util.DBXmlUtility.execute(DBXmlUtility.java:283)
        at com.adventnet.nms.db.util.DBXmlUtility.updatePropTable(DBXmlUtility.java:3180)
        at com.adventnet.nms.db.util.DBXmlUtility.modifyOneNode(DBXmlUtility.java:1467)
        at com.adventnet.nms.db.util.DBXmlUtility.modifyNode(DBXmlUtility.java:1437)
---
[20 May 2016 05:16:38:690] SYS_ERR: [WARNING] [c3p0] Another error has occurred [ org.postgresql.util.PSQLException: This connection has been closed. ] which will not be reported to listeners!org.postgresql.util.PSQLException: This connection has been closed.
        at org.postgresql.jdbc2.AbstractJdbc2Connection.checkClosed(AbstractJdbc2Connection.java:714)
        at org.postgresql.jdbc3.AbstractJdbc3Connection.createStatement(AbstractJdbc3Connection.java:230)
        at org.postgresql.jdbc2.AbstractJdbc2Connection.createStatement(AbstractJdbc2Connection.java:191)
        at com.mchange.v2.c3p0.impl.NewProxyConnection.createStatement(NewProxyConnection.java:96)
        at com.adventnet.nms.db.util.DBXmlUtility.execute(DBXmlUtility.java:283)
        at com.adventnet.nms.db.util.DBXmlUtility.getTableName(DBXmlUtility.java:3275)
        at com.adventnet.nms.db.util.DBXmlUtility.getAllAttributes(DBXmlUtility.java:2158)
        at com.adventnet.nms.commonfe.GenericFESession.update(GenericFESession.java:446)
        at com.adventnet.nms.commonfe.GenericFESession.update(GenericFESession.java:497)
        at java.util.Observable.notifyObservers(Unknown Source)
        at com.adventnet.nms.db.util.TreeEventNotifier.notifyObservers(TreeEventNotifier.java:59)
        at com.adventnet.nms.db.util.DBXmlUtility.notifyTheListeners(DBXmlUtility.java:494)
---
com.adventnet.security.authorization.AuthorizationException: An I/O error occured while sending to the backend. ; nested exception is: 
        com.adventnet.nms.store.NmsStorageException: An I/O error occured while sending to the backend., 
        Cause: 
        com.adventnet.nms.store.NmsStorageException: An I/O error occured while sending to the backend.
        at com.adventnet.security.authorization.RelationalAuthData.getOperationsForView(RelationalAuthData.java:1209)
        at com.adventnet.security.authorization.NortelAuthorizationEngineImpl.getOperationsForUser(NortelAuthorizationEngineImpl.java:613)
        at com.adventnet.nms.util.GenericUtility.getOperationsForUser(GenericUtility.java:461)
        at com.adventnet.nms.commonfe.GenericFESession.update(GenericFESession.java:462)
        at com.adventnet.nms.commonfe.GenericFESession.update(GenericFESession.java:497)
        at java.util.Observable.notifyObservers(Unknown Source)
        at com.adventnet.nms.db.util.TreeEventNotifier.notifyObservers(TreeEventNotifier.java:59)
        at com.adventnet.nms.db.util.DBXmlUtility.notifyTheListeners(DBXmlUtility.java:494)
        at com.adventnet.nms.db.util.DBXmlUtility.modifyNode(DBXmlUtility.java:1428)
        at com.adventnet.nms.mapdb.MapHandler.handleTreeUpdate(MapHandler.java:416)

ConnectionBreakHandlerImpl exceptions:
java.lang.NullPointerException
        at test.ConnectionBreakHandlerImpl.handleConnectionFail(ConnectionBreakHandlerImpl.java:139)
        at com.adventnet.management.transaction.ConnectionPool.handleConnectionFail(ConnectionPool.java:1586)
        at com.adventnet.management.transaction.ConnectionPool.dbClosed(ConnectionPool.java:1795)
        at com.adventnet.management.transaction.TransactionAPI.beginHbnTxn(TransactionAPI.java:947)
        at com.adventnet.management.transaction.TransactionAPI.beginTxn(TransactionAPI.java:851)
        at com.adventnet.management.transaction.TransactionAPI.begin(TransactionAPI.java:309)
        at com.adventnet.nms.store.relational.hbn.HbnTopoPersistence.beginTxn(HbnTopoPersistence.java:1129)
        at com.adventnet.nms.store.relational.hbn.HbnTopoPersistence.get(HbnTopoPersistence.java:312)
        at com.adventnet.nms.store.relational.hbn.HbnTopoPersistence.getObject(HbnTopoPersistence.java:250)
        at com.adventnet.nms.topodb.CommonTopoAPI.getObject(CommonTopoAPI.java:461)
        at com.adventnet.nms.topodb.TopoDB.getByName(TopoDB.java:3896)
        at com.adventnet.nms.topodb.TopoDB.getByName(TopoDB.java:3871)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenSender.getMoByName(SirenSender.java:64)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenSender.sendObjectChange(SirenSender.java:234)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenDevice.buildComponent(SirenDevice.java:1139)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenDevice.buildComponent(SirenDevice.java:1075)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenDevice.processEntity(SirenDevice.java:656)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenDevice.processReadedData(SirenDevice.java:575)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenDevice.processReadedResponse(SirenDevice.java:529)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenDevice.processReadedMsg(SirenDevice.java:478)
        at com.genband.gvm.server.deviceadapter.device.siren.SirenDevice.process(SirenDevice.java:319)
        at com.genband.gvm.server.deviceadapter.util.QueueWatcher.fireProcess(QueueWatcher.java:236)
        at com.genband.gvm.server.deviceadapter.util.QueueWatcher.run(QueueWatcher.java:203)
---
at org.postgresql.jdbc2.AbstractJdbc2Connection.checkClosed(AbstractJdbc2Connection.java:714)
at org.postgresql.jdbc3.AbstractJdbc3Connection.createStatement(AbstractJdbc3Connection.java:230)
at org.postgresql.jdbc2.AbstractJdbc2Connection.createStatement(AbstractJdbc2Connection.java:191)
at com.mchange.v2.c3p0.impl.NewProxyConnection.createStatement(NewProxyConnection.java:96)
at test.ConnectionBreakHandlerImpl.handleConnectionFail(ConnectionBreakHandlerImpl.java:203)
at test.ConnectionBreakHandlerImpl$1.run(ConnectionBreakHandlerImpl.java:172)

---
java.lang.NullPointerException
        at test.ConnectionBreakHandlerImpl.handleConnectionFail(ConnectionBreakHandlerImpl.java:124)
        at com.adventnet.management.transaction.ConnectionPool.handleConnectionFail(ConnectionPool.java:1586)
        at com.adventnet.management.transaction.ConnectionPool.executeQueryStmt(ConnectionPool.java:743)
---
org.postgresql.util.PSQLException: This connection has been closed.
        at org.postgresql.jdbc2.AbstractJdbc2Connection.checkClosed(AbstractJdbc2Connection.java:714)
        at org.postgresql.jdbc3.AbstractJdbc3Connection.createStatement(AbstractJdbc3Connection.java:230)
        at org.postgresql.jdbc2.AbstractJdbc2Connection.createStatement(AbstractJdbc2Connection.java:191)
        at com.mchange.v2.c3p0.impl.NewProxyConnection.createStatement(NewProxyConnection.java:96)
        at test.ConnectionBreakHandlerImpl.handleConnectionFail(ConnectionBreakHandlerImpl.java:203)
        at test.ConnectionBreakHandlerImpl$1.run(ConnectionBreakHandlerImpl.java:172)
